lib/matplotlib/tests/test_rcparams.py

```diff
@@ -592,8 +591,32 @@ def test_deprecation(monkeypatch):
     # suppress_matplotlib_deprecation_warning, rather than any explicit check.
 
 
-def test_rcparams_legend_loc():
-    value = (0.9, .7)
-    match_str = f"{value} is not a valid value for legend.loc;"
-    with pytest.raises(ValueError, match=re.escape(match_str)):
-        mpl.RcParams({'legend.loc': value})
+@pytest.mark.parametrize('value, allowed', [
+    ('best', True),
+    ('upper right', True),
+    ('upper left', True),
+    ('lower left', True),
+    ('lower right', True),
+    ('right', True),
+    ('center left', True),
+    ('center right', True),
+    ('lower center', True),
+    ('upper center', True),
+    ('center', True),
+    (1, True),
+    ((0.9, .7), True,),
+    ('(0.9, .7)', True),
```

Comment:
If we are going to make it take these values at run time in Python we also need to be able to put them in the matplotribrc file so if it is a string we need to take a try at the "sequence of floats" validator as well.

Reply:
I don't want to waste your time, but I'm not sure I fully understood what you're saying. Can you guide me a bit more?

Reply:
@tacaswell Please don't forget about me :)

Reply:
sorry, I get too many notifications.....


I mean that anything in the rcParams should be setabble from `matplotlibrc`.  Given that the matplotlibrc is a simple text format the values come to us ...as text so the validator (which is more of a converter, but that is a different topic), needs to correctly convert a sequence of floats the way that is encoded in `matplotlibrc`.

Reply:
All good, I had no doubt about it.
That helped a lot! thank you so much. I'll start working on it.